packages/xstate-solid/src/fsm.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  MachineImplementationsFrom,
+  ServiceFrom,
+  StateFrom,
+  StateMachine
+} from '@xstate/fsm';
+import { createMachine, interpret } from '@xstate/fsm';
+import type { Accessor } from 'solid-js';
+import { createMemo, onCleanup, createEffect } from 'solid-js';
+import { createImmutable } from './createImmutable';
+
+type UseFSMReturn<TService extends StateMachine.AnyService> = [
+  StateFrom<TService>,
+  TService['send'],
+  TService
+];
+
+export function useMachine<TMachine extends StateMachine.AnyMachine>(
+  machine: TMachine,
+  options?: MachineImplementationsFrom<TMachine>
+): UseFSMReturn<ServiceFrom<TMachine>> {
+  const resolvedMachine = createMachine(
+    machine.config,
+    options ? options : (machine as any)._options
+  );
+
+  const service = interpret(resolvedMachine).start();
```

Comment:
shouldn't we start this conditionally here in the same way we do it in here?
https://github.com/statelyai/xstate/blob/3addece3ba1803eed840e5586f46dfe57f1982ea/packages/xstate-solid/src/createService.ts#L37-L40

Reply:
+ isn't this missing a cleanup?

Reply:
similarly - we could have some tests for this cleanup thing here (it's probably not worth testing the browser/server paths separately, let's just focus on the browser case)